compiler/src/main/java/com/squareup/anvil/compiler/codegen/Parameter.kt

```diff
@@ -35,8 +36,31 @@ sealed interface Parameter {
     }
 }
 
+/**
+ * Returns a name which is unique when compared to the [Parameter.originalName] of the
+ * [superParameters] argument.
+ *
+ * This mimics Dagger's method of unique naming.  If there are three parameters named "foo", the
+ * unique parameter names will be [foo, foo2, foo3].
```

Comment:
I struggled with the description first and couldn't understand why it's needed without looking at the tests added in this PR. First, when I read "parameter" I was thinking of constructor injection, but there can't be multiple parameters with the same name. It would help to mention that this is related to member injection and the same parameter names come from injected members in super types.

Reply:
Great feedback, thanks!

Reply:
Perfect 👍🏻